tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/binaryoperations/EqualAndNotEqualOperationsTest.java

```diff
@@ -893,20 +898,20 @@ public void testEqualAndNotEqualNegativeCases() {
         validateError(resultNegative, i++, "operator '!=' not defined for 'int[]' and '[float,float]'", 82, 35);
         validateError(resultNegative, i++, "operator '==' not defined for 'int[]' and '[int,float]'", 85, 23);
         validateError(resultNegative, i++, "operator '!=' not defined for '[int,float]' and 'int[]'", 85, 35);
-        validateError(resultNegative, i++, "operator '==' not defined for '[int,map<int>]' and '[int,float]'", 99,
-                      23);
-        validateError(resultNegative, i++, "operator '!=' not defined for '[int,float]' and '[int,map<int>]'", 99,
-                      35);
-        validateError(resultNegative, i++, "operator '==' not defined for 'any' and 'int'", 103, 15);
-        validateError(resultNegative, i++, "operator '!=' not defined for 'int' and 'any'", 103, 27);
-        validateError(resultNegative, i++, "operator '==' not defined for 'map<(int|string)>' and 'map'", 107, 15);
-        validateError(resultNegative, i++, "operator '!=' not defined for 'map' and 'map<(int|string)>'", 107, 27);
-        validateError(resultNegative, i++, "operator '==' not defined for 'Employee' and '()'", 154, 9);
-        validateError(resultNegative, i++, "operator '==' not defined for 'Foo' and '()'", 160, 9);
+        validateError(resultNegative, i++, "operator '==' not defined for 'Employee' and '()'", 138, 9);
+        validateError(resultNegative, i++, "operator '==' not defined for 'Foo' and '()'", 144, 9);
         validateError(resultNegative, i++, "operator '==' not defined for 'function () returns (string)' and '()'",
-                      166, 9);
+                      150, 9);
         validateError(resultNegative, i++, "operator '!=' not defined for 'readonly' and 'map<int>'",
-                184, 12);
+                168, 12);
+        validateError(resultNegative, i++, "operator '==' not defined for '[int,map<int>]' and '[int,float]'", 179,
+                23);
+        validateError(resultNegative, i++, "operator '!=' not defined for '[int,float]' and '[int,map<int>]'", 179,
+                35);
+        validateError(resultNegative, i++, "operator '==' not defined for 'MyObject' and '()'", 182,
+                15);
+        validateError(resultNegative, i++, "operator '!=' not defined for 'MyObject' and '()'", 182,
```

Comment:
If we don't have a test already, let's add one where both operands are not anydata.